### ISSUE TITLE
Sandbox: opam root not visible

### ISSUE THREAD

**Issue by copy** [ocaml/opam#3434]
[ocb-stubblr](https://github.com/pqwy/ocb-stubblr) is an ocamlbuild plugin that invokes the opam binary during compilation. The Linux version of opam's sandbox script makes `$OPAM_SWITCH_PREFIX` visible, but not `$OPAMROOT`, so the ocp-stubblr's opam invokation can't detect the opam root (if it doesn't happen to be in `/home`):

```
ocamlfind ocamlopt -package unix -package ocamlbuild -linkpkg -package ocb-stubblr myocamlbuild.ml /media/samsung/opam2/4.06.1/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
[ERROR] /media/samsung/opam2 exists, but does not appear to be a valid opam root. Please remove it and use `opam init', or specify a different `--root' argument
Failure: Ocb_stubblr: error running opam.
```

**Comment by rjbou** [comment]
Related [mirage issue](https://github.com/mirage/mirage-platform/issues/200).
I reproduce my message here:
_During build, `OPAMROOT` is not binded. Usually it is in `HOME`, which is binded `ro`. At build stage, opam should not be invoked, especially with a command that could change opam state (that needs `OPAMROOT` to be `rw` binded). Needed variables can be found in the environment, defined/found in the opam file and transmitted, or set at this stage._

**Comment by rjbou** [comment]
Resolved via #3540. Thanks!
